src/interp.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2013 The Noise-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains methods for interpolating between points.
+use num::{Float};
+use math;
+
+/// Performs linear interploation between two values.
+pub fn linear<T: Float>(lhs: T, rhs: T, a: T) -> T {
+    ((T::one() - a) * lhs) + (a * rhs)
+}
+
+/// Maps a value onto a cubic S-curve.
+pub fn s_curve3<T: Float>(a: T) -> T {
+    a * a * (math::cast::<_, T>(3.0) - (math::cast::<_, T>(2.0) * a))
+}
+
```

Comment:
I dunno - still feel like it would be ok to put these in `math` for now, as opposed to prematurely putting them in a different module.